client/lib/sites-list/list.js

```diff
@@ -386,6 +388,42 @@ SitesList.prototype.isSelected = function( site ) {
 };
 
 /**
+ * Is the site starred?
+ *
+ * @api public
+ */
+SitesList.prototype.isStarred = function( site ) {
+	return this.starred.indexOf( site.ID ) > -1;
+};
+
+SitesList.prototype.toggleStarred = function( siteID ) {
+	if ( ! siteID ) {
+		return;
+	}
+
+	// do not add duplicates
+	if ( this.starred.indexOf( siteID ) === -1 ) {
+		this.starred.unshift( siteID );
+	} else {
+		this.starred.splice( this.starred.indexOf( siteID ), 1 );
+	}
+
+	PreferencesActions.set( 'starredSites', this.starred );
+
+	this.emit( 'change' );
+};
+
+SitesList.prototype.getStarred = function() {
+	this.starred = PreferencesStore.get( 'starredSites' ) || [];
+
+	if ( ! this.initialized ) {
```

Comment:
>It would try to operate on the array hitting missing attributes in different function calls on the components.

I'm under the impression that the array would never _not_ be empty, if the sites list hasn't yet been initialized?